File: back/go/app/core/data/repository/GameRepository.go
package repository

import (
	"context"
	"go-ttt/app/core/database/redis/entity"
	"go-ttt/app/core/model"
)

type GameRepository interface {
	CreateNewGame(ctx context.Context, newKey string, board *model.StateType, player *model.PlayerType) (*model.GameType, error)
	ResetGame(ctx context.Context, gameID string, board *model.StateType, player *model.PlayerType) (*model.GameType, error)
	UpdateBoard(ctx context.Context, gameID string, position *model.CellPosition, player *model.PlayerType) (*model.StateType, error)
	GetCurrentPlayer(ctx context.Context, gameID string) (*model.PlayerType, error)
	GetBoardState(ctx context.Context, gameID string) (*model.StateType, error)
	GetGameState(ctx context.Context, gameID string) (*model.GameType, error)
	UpdateGameState(ctx context.Context, gameID string, board *model.StateType, info *entity.GameInfo) (*model.GameType, error)
}

File: back/swift/app/core/data/repository/GameRepository.swift
import Foundation

protocol GameRepository {
    func createNewGame(newKey: String, board: BoardType, player: PlayerType) -> Result<GameType, Error>
    func resetGame(gameID: String, board: BoardType, player: PlayerType) -> Result<GameType, Error>
    func updateBoard(gameID: String, position: CellPosition, player: PlayerType) -> Result<StateType, Error>
    func getCurrentPlayer(gameID: String) -> Result<PlayerType, Error>
    func getBoardState(gameID: String) -> Result<StateType, Error>
    func getGameState(gameID: String) -> Result<GameType, Error>
}